src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBDatabaseToRestore.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBDatabaseToRestore"), OutputType(typeof(PSDatabaseToRestore))]
+    public class NewAzCosmosDBDatabaseToRestore : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = Constants.RestoreDatabaseNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.RestoreCollectionNamesHelpMessage)]
+        public string[] CollectionName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            PSDatabaseToRestore databaseToRestore = new PSDatabaseToRestore(DatabaseName, CollectionName);
+            WriteObject(databaseToRestore);
+
+            return;
+        }
+    }
+}
```

Comment:
Do you really need this separate cmdlet ? You can just ask the user to provide the parameters, CollectionNAme  and DatabaseName, in place of DatabaseToRestore parmeter (wherever you are using it)

Reply:
The user can provide multiple databases. And each database has multiple collections. 
So its not possible to give multiple objects in a clean way for powerhsell.